### ISSUE TITLE
The very first ON TIMER(n) statement gets trapped immediately

### ISSUE THREAD

**Issue by robhagemans** [robhagemans/pcbasic#25]
Reported by Francis Lee https://sourceforge.net/p/pcbasic/discussion/bugs/thread/8249e7f7/

I was learning and testing ON KEY / ON TIMER statement, and I believe I found a bug in ON TIMER(n) stetement implementation. Below is the code I was experimenting with:
```
10 CLS
15 PRINT "Press UP arrow key to exit...
20 ON KEY(11) GOSUB 80
30 KEY(11) ON
32 ON TIMER(1) GOSUB 100
34 TIMER ON
40 FOR I=1 TO 10
50 PRINT "Hello, World!
60 NEXT I
70 GOTO 70
80 KEY(11) OFF:PRINT "Removing KEY(11) trap setting...
90 END
100 PRINT "Hello, every one second!
110 RETURN 70 'go to infinite loop
```
I was expecting "Hello, World!" to be printed first 10 times, followed by "Hello, every one second!" later every one second, however, "Hello, World!" never gets printed at all. "Hello, every one second!" gets printed immediately after TIMER ON statement. BTW, print statement without the terminating double quote was written intentionally in such style after learning that LOAD/SAVE/CHDIR statement works fine like.

**Comment by robhagemans** [comment]
I also found that KEY(11) OFF is not required in GW-BASIC 3.23 (with dosbox) when the program is terminiated, however without this statement, the UP arrow key is not responding anymore in PC-BASIC.

**Comment by robhagemans** [comment]
Additionally, in 16.12, the up arrow is not trapped

**Comment by robhagemans** [comment]
Primary issue (TIMER trigger) fixed by commits 4df8393 (16.12) and 8b7a4c3 (15.08)

**Comment by robhagemans** [comment]
16.12 remaining key trapping issue spun off as issue #33

**Comment by robhagemans** [comment]
Non-responding up arrow (15.08) fixed by commit eebeaea
